File: packages/sdk-py/src/grantex/resources/_principal_sessions.py
from __future__ import annotations

from .._http import HttpClient
from .._types import CreatePrincipalSessionParams, PrincipalSessionResponse


class PrincipalSessionsClient:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    def create(self, params: CreatePrincipalSessionParams) -> PrincipalSessionResponse:
        data = self._http.post("/v1/principal-sessions", params.to_dict())
        return PrincipalSessionResponse.from_dict(data)

File: packages/sdk-ts/src/resources/principal-sessions.ts
import type { HttpClient } from '../http.js';
import type { CreatePrincipalSessionParams, PrincipalSessionResponse } from '../types.js';

export class PrincipalSessionsClient {
  readonly #http: HttpClient;

  constructor(http: HttpClient) {
    this.#http = http;
  }

  create(params: CreatePrincipalSessionParams): Promise<PrincipalSessionResponse> {
    return this.#http.post<PrincipalSessionResponse>('/v1/principal-sessions', params);
  }
}
